smartdashboard/views.py

```diff
@@ -342,3 +343,203 @@ def __init__(
         self.app_view = app_view
         self.ens_view = ens_view
         self.orc_view = orc_view
+
+
+class MemoryView(ViewBase):
+    """View class for memory section of the Database Telemetry page"""
+
+    def __init__(self, shard: t.Optional[Shard]) -> None:
+        """Initialize a MemoryView
+
+        :param shard: Selected shard
+        :type shard: t.Optional[Shard]
+        """
+        self.shard = shard
+        self.memory_table_element = DeltaGenerator()
+        self.memory_graph_element = DeltaGenerator()
+
+    def update_shard(self, new_shard: t.Optional[Shard]) -> None:
```

Comment:
It may be asking too much, but consider breaking each of these memory stats into individual classes... It might be nice if _each_ dynamic view has the same API

e.g. If each one has the same contract (an `update` method vs update_x, update_y, and update) it becomes easier to use each component and add new ones.

Reply:
Each dynamic view has an `update` method, but some of the views have different components it needs to update. Within each `update` method there could be `update_logs`, `update_status`, etc. In particular, this `update_shard` method is called when a user selects a new shard, so the view's shard needs to be fully switched out. I'm going to close this as resolved for now but I do think there's refactoring work that needs to be done in the near future that will make all of this hopefully way more clean and understandable!